code/game/objects/items/devices/gps.dm

```diff
@@ -17,15 +18,98 @@ var/list/GPS_list = list()
 	var/hide_signal = FALSE		// If true, signal is not visible to other GPS devices.
 	var/can_hide_signal = FALSE	// If it can toggle the above var.
 
+	var/mob/holder
+	var/list/tracking_devices
+	var/obj/compass_holder/compass
+
 /obj/item/device/gps/Initialize()
 	. = ..()
+	compass = new(src)
 	GPS_list += src
 	name = "global positioning system ([gps_tag])"
+	update_holder()
 	update_icon()
 
+/obj/item/device/gps/proc/update_holder()
+
+	if(holder && loc != holder)
+		GLOB.moved_event.unregister(holder, src)
+		GLOB.dir_set_event.unregister(holder, src)
+		holder.client?.screen -= compass
+		holder = null
+
+	if(istype(loc, /mob))
+		holder = loc
+		GLOB.moved_event.register(holder, src, .proc/update_compass)
+		GLOB.dir_set_event.register(holder, src, .proc/update_compass)
+
+	if(holder && tracking)
+		if(!(src in SSobj.processing))
+			START_PROCESSING(SSobj, src)
```

Comment:
It's purely a decision by whoever implemented it, presumably because `|=` is slower than `+=` (particularly on enormous lists like the processing systems end up with). I could copy START_PROCESSING to add one with `|=` if it would be clearer.

Reply:
Decided to track it on the object with a bool since that seems to be the least potentially disruptive solution.